src/cbor/CBOREncoder.cpp

```diff
@@ -45,24 +46,68 @@ CborError CBOREncoder::encode(PropertyContainer & property_container, uint8_t *
    */
   CborError error = CborNoError;
   int num_encoded_properties = 0;
-  std::for_each(property_container.begin(),
-                property_container.end(),
-                [lightPayload, &arrayEncoder, &error, &num_encoded_properties](Property * p)
-                {
-                  if (p->shouldBeUpdated() && p->isReadableByCloud())
-                  {
-                    error = p->append(&arrayEncoder, lightPayload);
-                    if(error == CborNoError)
-                      num_encoded_properties++;
-                    else
-                      return;
-                  }
-                });
-  if ((CborNoError != error) && 
-      (CborErrorOutOfMemory != error))
+  int num_checked_properties = 0;
+  static int encoded_properties_message_limit = CBOR_ENCODER_NO_PROPERTIES_LIMIT;
+
+  if(current_property_index >= property_container.size())
+    current_property_index = 0;
+
+  PropertyContainer::iterator iter = property_container.begin();
+  std::advance(iter, current_property_index);
+
+  for(; iter != property_container.end(); iter++)
+  {
+    Property * p = * iter;
+    bool maximum_number_of_properties_reached = (num_encoded_properties >= encoded_properties_message_limit) && (encoded_properties_message_limit != -1);
```

Comment:
```suggestion
    bool const maximum_number_of_properties_reached = (num_encoded_properties >= encoded_properties_message_limit) && (encoded_properties_message_limit != -1);
```
Use [const-correctness](https://isocpp.org/wiki/faq/const-correctness#overview-const) as much as possible :wink:

Reply:
Fixed. https://github.com/arduino-libraries/ArduinoIoTCloud/pull/293/commits/eedffb304d4fd5a1890e7f030080a5f23d49cad3